lib/sycamore/sycamore/llms/llms.py

```diff
@@ -23,9 +23,16 @@ class LLMMode(Enum):
 class LLM(ABC):
     """Abstract representation of an LLM instance. and should be subclassed to implement specific LLM providers."""
 
-    def __init__(self, model_name, cache: Optional[Cache] = None):
+    def __init__(self, model_name, cache: Optional[Cache] = None, default_mode: Optional[LLMMode] = None):
```

Comment:
make default_mode not optional here.  You could do  = LLMMode.SYNC, but given how few subclasses we have I think it's better to have it be explicit.